From: Vlastimil Babka <vbabka@suse.cz>
Subject: fixup! mm, slab: move memcg charging to post-alloc hook
Date: Wed, 3 Apr 2024 17:45:15 +0200

The call to memcg_alloc_abort_single() is wrong, it expects a pointer to
single object, not an array.

Link: https://lkml.kernel.org/r/4af50be2-4109-45e5-8a36-2136252a635e@suse.cz
Signed-off-by: Vlastimil Babka <vbabka@suse.cz>
Reported-by: Aishwarya TCV <aishwarya.tcv@arm.com>
Tested-by: Aishwarya TCV <aishwarya.tcv@arm.com>
Reviewed-by: Roman Gushchin <roman.gushchin@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/slub.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/slub.c~mm-slab-move-memcg-charging-to-post-alloc-hook-fix
+++ a/mm/slub.c
@@ -2172,7 +2172,7 @@ bool memcg_slab_post_alloc_hook(struct k
 		return true;
 
 	if (likely(size == 1)) {
-		memcg_alloc_abort_single(s, p);
+		memcg_alloc_abort_single(s, *p);
 		*p = NULL;
 	} else {
 		kmem_cache_free_bulk(s, size, p);
_
